### PR TITLE
Named vector multi search

### PR DESCRIPTION
**Pull request by dirkkul**

### CODE REVIEW

File: weaviate/collections/grpc/query.py [hunk 1]
```diff
@@ -301,22 +322,97 @@ def near_vector(
...
+                raise WeaviateInvalidInputError(
+                    "The number of target vectors must be equal to the number of vectors."
+                )
+
+            vector_per_target: Dict[str, bytes] = {}
+            for key, value in near_vector.items():
+                if (
+                    not isinstance(value, list)
+                    or len(value) == 0
+                    or not isinstance(value[0], get_args(NUMBER))
```
Comment (tsmith023) [inline weaviate/collections/grpc/query.py:355]: I've not seen this pattern before. Does it allow the instance of `value[0]` to be checked against both `int` and `float` within the same `isinstance` call?

File: integration/test_named_vectors.py [hunk 2]
```diff
@@ -562,3 +560,85 @@ def test_duplicate_named_vectors(collection_factory: CollectionFactory) -> None:
...
+        vectorizer_config=[
+            wvc.config.Configure.NamedVectors.none("first"),
+            wvc.config.Configure.NamedVectors.none("second"),
+        ],
+    )
+
+    uuid1 = collection.data.insert({}, vector={"first": [1, 0, 0], "second": [0, 1, 0]})
+    uuid2 = collection.data.insert({}, vector={"first": [0, 1, 0], "second": [1, 0, 0]})
+
+    objs = collection.query.near_vector([1.0, 0.0, 0.0], target_vector=["first", "second"]).objects
```
Comment (tsmith023) [inline integration/test_named_vectors.py:594]: The `target_vector` parametrized argument isn't being used here

**Review by tsmith023 — CHANGES_REQUESTED** [review]
Just the test needs changing, the other comment about variadics is just a consideration/thought 😁

File: integration/test_named_vectors.py (hunk 2 above)
Comment (dirkkul) [reply integration/test_named_vectors.py]: thx! Fixed now - I would expect a linter to warn there

File: weaviate/collections/grpc/query.py (hunk 1 above)
Comment (dirkkul) [reply weaviate/collections/grpc/query.py:355]: I think chatGPT proposed it 😅 

> Does it allow the instance of value[0] to be checked against both int and float within the same isinstance call

Yes, AFAIK this just works in newer python versions, but not the old stuff we still have to support

File: weaviate/collections/classes/grpc.py [hunk 3]
```diff
@@ -246,6 +247,87 @@ class _HybridNearVector(_HybridNearBase):
...
+        return _MultiTargetVectorJoin(
+            combination=_MultiTargetVectorJoinEnum.AVERAGE, target_vectors=target_vectors
+        )
+
+    @staticmethod
+    def minimum(target_vectors: List[str]) -> _MultiTargetVectorJoin:
+        """Combine the distance from different target vectors by using the minimum distance."""
+        return _MultiTargetVectorJoin(
+            combination=_MultiTargetVectorJoinEnum.MINIMUM, target_vectors=target_vectors
+        )
```
[the remark this answers is not in the document]
Comment (dirkkul) [reply weaviate/collections/classes/grpc.py:310]: hmmm - I think I prefer the lists. We do not use variadics anywhere else and quick googling didn't show any possibility to accept both. But nice idea :)

**Review by tsmith023 — APPROVED** [review]

**Pull request merged**